src/Agent/NewRelic/Agent/Core/Segments/NoOpSegment.cs

```diff
@@ -40,6 +40,7 @@ public class NoOpSegment : ISegment, ISegmentExperimental, ISegmentDataState
 		
         public void End() { }
         public void End(Exception ex) { }
+        public void EndStackExchangeRedis() { }
```

Comment:
In general I agree, but since this is a bug fix I think we probably want to limit how much refactoring we do. Definitely something to keep in mind, though. We can also add a tech debt story to the backlog to revisit.